File: pset6/exercises/readability/readability.py
from cs50 import get_string

def countLetters(digit):

    if digit.isalpha(): return 1
    return 0

def countSentences(digit):

    digits = [".", "!", "?"]
    if digit in digits: return 1
    return 0

def countWords(digit):

    if digit.isspace(): return 1
    return 0

def printGrade(words,letters,sentences):

    l = letters / words * 100
    s = sentences / words * 100

    i = int(0.0588 * l - 0.29 * s - 15.8)

    if i > 16:
        print("Grade 16+")

    elif i > 1 and i < 17:
        print(f"Grade {i}")

    else:
        print("Before Grade 1")

text = get_string("Text: ")
size = len(text)

words,letters,sentences = 0, 0, 0

for i in range(size):

    words += countWords(text[i])
    letters += countLetters(text[i])
    sentences += countSentences(text[i])

printGrade(words + 1,letters,sentences)

File: pset2/exercises/readability/readability.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>
#include <ctype.h>
#include <string.h>

int CountWords(char digit);
int CountLetters(char digit);
int CountSentences(char digit);
int Calculate(int letters, int words, int sentences);

int main(void)
{
    string text = get_string("Text: ");
    int letters, words, sentences = 0;

    for (int i = 0, n = strlen(text); i < n; i++)
    {
        // Estas funções verificam se digito é um espaço, uma letra ou o final de uma frase o contabilizam.
        words = words + CountWords(text[i]);
        letters = letters + CountLetters(text[i]);
        sentences = sentences + CountSentences(text[i]);
    }

    // Somo 1 as palavras para contabilizar a ultima palavra do texto.
    words = words + 1;

    int indice = Calculate(letters, words, sentences);

    if (indice >= 16)
    {
        printf("Grade 16+\n");
    }
    else if (indice < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", indice);
    }

}

int CountWords(char digit)
{
    int digits = 0;

    if (isspace(digit))
    {
        digits = digits + 1;
    }

    return digits;
}

int CountLetters(char digit)
{
    int digits = 0;

    if (isalpha(digit))
    {
        digits = digits + 1;
    }

    return digits;
}

int CountSentences(char digit)
{
    int digits = 0;

    if (digit == '.' || digit == '?' || digit == '!')
    {
        digits = digits + 1;
    }

    return digits;
}

int Calculate(int letters, int words, int sentences)
{
    double L = ((float)letters / (float)words) * 100;
    double S = ((float)sentences / (float)words) * 100;

    double FloatIndex = 0.0588 * L - 0.296 * S - 15.8;
    int IntIndex = round(FloatIndex);

    return IntIndex;
}